tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -174,6 +174,26 @@ void AnnotateThunkLaunchDimensions(const LaunchDimensions& launch_dims,
       {llvm::ConstantAsMetadata::get(ir_kernel),
        llvm::MDString::get(llvm_context, "reqntidx"),
        llvm::ConstantAsMetadata::get(threads_per_block_ir_value)}));
+  if (launch_dims.thread_counts_per_block().y > 1) {
```

Comment:
Could we extract the logic which finds `threads_per_block_ir_value` into a separate function returning `llvm::Value`? Also while we are there could you write a comment describing the logic, I'm not sure I quite follow? (E.g. why do we prefer z-dimension if both z- and y- are greater than 1? Or that never happens?)

Reply:
I refactored.
This doesn't prefer anything. This just add annotations. The dimensions sizes are defined elsewhere.

Here, `threads_per_block_ir_value` was just reused to hold a temporary variable. The refactoring got rid of it.